### PR TITLE
Port `zelbet` to Plonkup

### LINKED ISSUE #9: Port zelbet to PLONKup
**Describe what you want implemented** 
Current code of zelbet should compile using the "stable" version of PlonkUp

**Describe "Why" this is needed**
Otherwise it won't compile unless it's using a custom and wrong version of PlonkUp.

**Describe alternatives you've considered**
Using a custom and wrong version of PlonkUp.

**Additional context**
N/A

### LINKED ISSUE #17: Add in-circuit decomposition for `bar`
**Describe what you want implemented** 
Decompose the witness using the s_i base values. This was previously done in plonkup [here](https://github.com/dusk-network/plonk/blob/v0.8.1-plonkup.0/src/constraint_system/zelbet.rs)

**Describe "Why" this is needed**
it is needed by in-circuit `bar`

**Describe alternatives you've considered**
NA

**Additional context**
Parts of the functionality seem to have moved to [TurboComposer::component_decomposition()](https://docs.rs/dusk-plonk/latest/dusk_plonk/prelude/struct.TurboComposer.html#method.component_decomposition) but the actual decomposing of the Scalar using the s_i base values seem to have gone missing. Since it is part of the `zelbet` algorithm, it makes sense to implement it in `zelbet` and not (as previously) in PLONK.

### PR DESCRIPTION
**Pull request by moCello**

### CODE REVIEW

File: src/hashing/gadgets.rs [hunk 1]
```diff
@@ -4,217 +4,257 @@
...
+                );
+                intermediate = u0;
+                remainder = u1 as u64;
+            }
+            false => remainder = intermediate[0],
+        }
+
+        reduced_decomposed[k] = remainder;
+        witness_decomposed[k] =
+            composer.append_witness(BlsScalar::from(remainder));
```
Comment (vlopes11) [inline src/hashing/gadgets.rs:146]: Put a comment on why this `append_witness` is secure (e.g. why a malicious prover cannot append arbitrary witnesses to create a valid proof with invalid arguments)

**Review by xevisalle — APPROVED** [review]

File: src/hashing/gadgets.rs (hunk 1 above)
Comment (moCello) [reply src/hashing/gadgets.rs:146]: I will look at it in #33

Comment (moCello) [reply src/hashing/gadgets.rs:146]: I need to create a new issue for it to avoid massive rebase

**Review by vlopes11 — APPROVED** [review]
LGTM

**Pull request merged**